grammar.js

```diff
@@ -573,6 +570,50 @@ module.exports = grammar({
       )
     },
 
+    // Identifier-ish, but useful enough to be their own nodes
+    dots: $ => "...",
+    dot_dot_i: $ => token(withPrec(PREC.DOT_DOT_I, /[.][.]\d+/)),
+
+    // NOTE: Technically R allows `...` and `..1` anywhere we want an `$.identifier`,
+    // but practically it can be useful for downstream consumers to have separate
+    // nodes for these particular constructs. Our compromise is to keep these as separate
+    // nodes, but then use this in most places we want an identifier.
+    _identifier_or_dots_or_dot_dot_i: $ => choice(
+      $.identifier,
+      $.dots,
+      $.dot_dot_i
+    ),
+
+    // NOTE: Having this as an actual node (rather than inlining the `choice()`) somehow
+    // ends up allowing better error recovery in a few cases
+    _string_or_identifier_or_dots_or_dot_dot_i: $ => choice(
+      $.string,
+      $.identifier,
+      $.dots,
+      $.dot_dot_i
+    ),
+
+    // NOTE: This is exactly `_string_or_identifier_or_dots_or_dot_dot_i` but with
+    // a precedence of 1. It seems like we have to set the `prec(1, )` on the `choice()`
+    // directly, we can't reuse `_string_or_identifier_or_dots_or_dot_dot_i` here
+    // otherwise `tree-sitter generate` throws an unresolved conflict error.
+    //
+    // This is only for use in `_argument_named`.
+    //
+    // Since `_argument_unnamed` can be an arbitrary `_expression` (with precedence 0)
+    // which includes `string`, `identifier`, `dots`, and `dot_dot_i`, there is an
+    // ambiguity between:
+    // - Starting the `value` of an `_argument_unnamed`
+    // - Starting the `name` of an `_argument_named`
+    //
+    // We set a higher precedence here to try and match `_argument_named` first.
+    _argument_name_string_or_identifier_or_dots_or_dot_dot_i: $ => prec(1, choice(
+      $.string,
+      $.identifier,
+      $.dots,
+      $.dot_dot_i
+    )),
```

Comment:
Slightly tricky case where the `"name"` field of `_argument_named` is technically `_string_or_identifier_or_dots_or_dot_dot_i`, but we need to bump the precedence there to disambiguate it from `_argument_unnamed`.

This replaces the `prec.right(1, )` that has been removed from `_argument_named`